### PR TITLE
refactor: split packages

### PR DESCRIPTION
**Pull request by keroxp**
This PR includes various changes across the package. Mainly, the most part of `cage` were split into sub packages for unit test convenience.

- `task`: codes for canary test, both ALB test and simple wait test
- `taskset`: an abstraction of combined tasks for single service
- `rollout`: a core logic part of `cage rollout`
- `types`: common interfaces
- `env`: environment variables

## logos/di introduction

We are now using logos/di for dependency management.

## Rewritten tests

Almost all tests have been rewritten, based on the policy below:
- detailed unit tests with gomock for sub packages, no `test` pacgaes
- integration tests with `test` package and failure tests with gomock for `cage` package

Now, I chose the same package for unit test codes not `_test`: to simulate failures that depends on internal state

## Other changes

- all `cage` commands skip loading task definition file if `--taskDefinitionArn` opt provided.
- The default value for `--canaryIdleDuration` changed to 15 seconds.
- fixed missing condition check that canary task has two or more attachments except `ElasticNetworkInterface`

### CODE REVIEW

File: key/keys.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package key
+
+type DepsKey string
+
+const (
+	EcsCli      DepsKey = "ecs"
```
[the remark this answers is not in the document]
Comment (keroxp) [reply key/keys.go:6]: There is the clear reason to use string key. If using int key, we'll obtain no information about key on missing dependency error, as it's a only serial number.

[the remark this answers is not in the document]
Comment (takkyuuplayer) [reply key/keys.go:6]: oh, i see 🙆‍♂️

File: export_test.go [hunk 2]
```diff
@@ -1,3 +1,9 @@
 package cage
 
+import "github.com/loilo-inc/logos/di"
+
 type CageExport = cage
+
+func NewCageExport(di *di.D) *cage {
+	return &cage{di}
+}
```
[the remark this answers is not in the document]
Comment (keroxp) [reply export_test.go:9]: Indeed.

**Review by alfnets — COMMENTED** [review]
In my review, I've only noted one point. The changes have made the errors and process more understandable. The adoption of dependency injection has contributed to streamlining the code. Nice work!

**Review by alfnets — APPROVED** [review]
LGTM! 🚀

**Review by takkyuuplayer — APPROVED** [review]
LGTM!

**Pull request merged**